File: day3/puzzle1.py
x = 0
y = 0
houses = {}
with open('day3.txt') as f:
 while True:
  address = ':'.join([str(x),str(y)])
  presents = 0
  if houses.has_key(address):
   presents = houses[address]
  houses[address] = presents + 1
  c = f.read(1)
  if not c:
   break
  if c == '^':
   y = y - 1
  if c == 'v':
   y = y + 1
  if c == '>':
   x = x + 1
  if c == '<':
   x = x - 1
print len(houses)

File: 2015/03/puzzle1.js
var fs = require('fs');

fs.readFile('day3.txt', 'utf-8', function (err, inp) {
 if (err) throw err;

 var x = 0, y = 0, houses = {};
 var dirs = { '^': [0,-1], 'v': [0,1], '>': [1,0], '<': [-1,0] };
 inp.trim().split("").forEach(function (c) {
  var address = x + ":" + y;
  houses[address] = (houses[address] || 0) + 1;
  var dir = dirs[c];
  x += dir[0]; y += dir[1];
 });
 console.log(Object.keys(houses).length);
});
